langchain/document_loaders/fauna.py

```diff
@@ -0,0 +1,56 @@
+from typing import List, Optional
+from langchain.docstore.document import Document
+
+from langchain.document_loaders.base import BaseLoader
+
+
+class FaunaLoader(BaseLoader):
+    def __init__(
+        self,
+        query: str,
+        page_content_field: str,
+        secrect: str,
+        metadata_fields: Optional[List[str]] = None,
+        after: Optional[str] = None,
+    ):
+        self.query = query
+        self.page_content_field = page_content_field
+        self.secrect = secrect
+        self.metadata_fields = metadata_fields
+
+    def load(self) -> List[Document]:
+        try:
+            from fauna import fql, Page
+            from fauna.client import Client
+            from fauna.encoding import QuerySuccess
+        except ImportError:
+            raise ImportError(
+                "Could not import fauna python package. "
+                "Please install it with `pip install fauna`."
+            )
+        documents = []
+        # Create Fauna Client
+        client = Client(secret=self.secrect) 
+        # Run FQL Query
+        response: QuerySuccess = client.query(fql(self.query))
+        page: Page = response.data
+        for result in page:
+            if result is not None:
+                document_dict = dict(result.items())
+                page_content = ''
+                for key, value in document_dict.items():
+                    if key == self.page_content_field:
+                        page_content = value
+                document: Document = Document(
+                    page_content=page_content,
+                    metadata={"id": result.id, "ts": result.ts},
+                )
+                documents.append(document)
+        if page.after is not None:
```

Comment:
is this saying there's pagination? like if you performed the query again you'd get more/new documents?

Reply:
Yes this is pagination. Following is an example how to use it. I have also added this in the docs in my latest commit.

```python
query = """
Item.paginate("hs+DzoPOg ... aY1hOohozrV7A")
Item.all()
"""
loader = FaunaLoader(query, field, secret)
```

You get a `after` value if there are more data. You can get values after the curcor by passing in the `after` string in query. 

To learn more following [this link](https://fqlx-beta--fauna-docs.netlify.app/fqlx/beta/reference/schema_entities/set/static-paginate)

Reply:
link seems to suggest you could just do (replace Item -> Set, drop Item.all())
```python
query = """
Set.paginate("hs+DzoPOg ... aY1hOohozrV7A")
"""
```
is that correct?

Reply:
Yes you are correct. Updated the docs